cmd/result-aggregator/main.go

```diff
@@ -110,15 +116,43 @@ func withErrorRate(request *results.Request) {
 	errorRate.With(labels).Inc()
 }
 
-func handleCIOperatorResult(username string, password func() []byte) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		start := time.Now()
+type validator interface {
+	Validate(username, password string) bool
+}
+
+func newValidator(username string, password string, htpasswdFileRaw string) (validator, error) {
+	var htpasswdFile *HtpasswdFile
+	if htpasswdFileRaw != "" {
```

Comment:
nit: a method that takes a list of inputs and uses `if` to do something different with each is a code smell

just call different methods originally

Reply:
Did not get rid of it completely.
Should look better after removing the u/p flag in the future.